src/common/sr_data_structs.c

```diff
@@ -561,7 +561,7 @@ sr_free_lock_item(void *lock_item)
 {
     CHECK_NULL_ARG_VOID(lock_item);
     sr_lock_item_t *li = (sr_lock_item_t *) lock_item;
-    free(li->filename);
+    free((void*)li->filename);
```

Comment:
Are you sure about this one? That might make it possible to assign a literal constant to ``filename`` and ``free()`` it later on.